bg_atlasgen/validate_atlases.py

```diff
@@ -0,0 +1,124 @@
+        "structures.json",
+        "meshes",
+    ]
+    for expected_file_name in expected_files:
+        expected_path = Path(atlas_path / expected_file_name)
+        assert (
+            expected_path.exists()
+        ), f"Expected file not found at {expected_path}"
+    return True
+
+
+def _assert_close(mesh_coord, annotation_coord, pixel_size):
+    """Helper function to check if the mesh and the annotation coordinate are closer to each other than 10 times the pixel size"""
+    assert (
+        abs(mesh_coord - annotation_coord) <= 10 * pixel_size
+    ), f"Mesh coordinate {mesh_coord} and annotation coordinate {annotation_coord} differ by more than 10 times pixel size {pixel_size}"
+    return True
+
+
+def validate_mesh_matches_image_extents(atlas: BrainGlobeAtlas):
+    """Checks if the mesh and the image extents are similar"""
+
+    root_mesh = atlas.mesh_from_structure("root")
+    annotation_image = atlas.annotation
+    resolution = atlas.resolution
+
+    z_range, y_range, x_range = np.nonzero(annotation_image)
+    z_min, z_max = np.min(z_range), np.max(z_range)
+    y_min, y_max = np.min(y_range), np.max(y_range)
+    x_min, x_max = np.min(x_range), np.max(x_range)
+
+    mesh_points = root_mesh.points
+    z_coords, y_coords, x_coords = (
+        mesh_points[:, 0],
+        mesh_points[:, 1],
+        mesh_points[:, 2],
+    )
+    z_min_mesh, z_max_mesh = np.min(z_coords), np.max(z_coords)
+    y_min_mesh, y_max_mesh = np.min(y_coords), np.max(y_coords)
+    x_min_mesh, x_max_mesh = np.min(x_coords), np.max(x_coords)
+
+    z_min_scaled, z_max_scaled = z_min * resolution[0], z_max * resolution[0]
+    y_min_scaled, y_max_scaled = y_min * resolution[1], y_max * resolution[1]
+    x_min_scaled, x_max_scaled = x_min * resolution[2], x_max * resolution[2]
+
+    _assert_close(z_min_mesh, z_min_scaled, resolution[0])
+    _assert_close(z_max_mesh, z_max_scaled, resolution[0])
+    _assert_close(y_min_mesh, y_min_scaled, resolution[1])
+    _assert_close(y_max_mesh, y_max_scaled, resolution[1])
+    _assert_close(x_min_mesh, x_min_scaled, resolution[2])
+    _assert_close(x_max_mesh, x_max_scaled, resolution[2])
+
+    return True
+
+
+def open_for_visual_check(atlas):
+    pass
+
+
+def validate_checksum(atlas):
+    pass
+
+
+def check_additional_references():
+    # check additional references are different, but have same dimensions
+    pass
+
+
+def validate_atlas(atlas_name, version):
+    """Validates the latest version of a given atlas"""
+
+    print(atlas_name, version)
+    atlas = BrainGlobeAtlas(atlas_name)
+    updated = get_atlases_lastversions()[atlas_name]["updated"]
+    if not updated:
+        update_atlas(atlas_name)
+    atlas_path = Path(get_brainglobe_dir()) / f"{atlas_name}_v{version}"
+    assert validate_atlas_files(
+        atlas_path
+    ), f"Atlas file {atlas_path} validation failed"
+    assert validate_mesh_matches_image_extents(
+        atlas
+    ), "Atlas object validation failed"
```

Comment:
"Atlas object validation" is too abstract for this message. You didn't check the entire object, just the extents of the annotation image and the mesh, right?

Reply:
Unless this message anticipates the addition of more checks in this group